### PR TITLE
add rest showdishes api

### PR DESCRIPTION
**Pull request by SoniaChoo**

### CODE REVIEW

File: db/rest.go [hunk 1]
```diff
@@ -26,3 +26,37 @@ func InsertRest(r *model.Rest) error {
 
 	return nil
 }
+
```
Comment (ziyitony) [inline db/rest.go:29]: please don't leave so much empty space

File: db/rest.go [hunk 2]
```diff
@@ -26,3 +26,37 @@ func InsertRest(r *model.Rest) error {
...
+		temp := &model.Dish{}
+		if err = row.Scan(&temp.DishId, &temp.RestId, &temp.Price, &temp.DishName, &temp.Stock, &temp.Sales, &temp.CreatTime); err != nil{
+			log.Printf("doing show dishes record loop with error %s\n", err.Error())
+			return nil, err
+		}
+		dishes = append(dishes, temp)
+	}
+	return dishes, nil
+}
+
```
Comment (ziyitony) [inline db/rest.go]: same here

File: http/restHandler.go [hunk 3]
```diff
@@ -53,3 +53,35 @@ func RegisterHandlerRest(w http.ResponseWriter, r *http.Request) {
...
+	//search all dishes belong to the rest
+	dishes, err := db.ShowDishesRest(&rest)
+	if  err != nil {
+		log.Printf("show dishes failed, error is %s\n", err.Error())
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "show dishes failed, can't show this rest's dishes, error is %s\n", err.Error())
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	fmt.Fprintf(w, "all dishes of rest %v are successfully showed as followed: \n%v", rest.RestId, dishes)
```
Comment (ziyitony) [inline http/restHandler.go]: `followed` -> `following`

`\n%v` -> `%v\n`

File: db/rest.go (hunk 1 above)
Comment (SoniaChoo) [reply db/rest.go:29]: ok, updated.

File: db/rest.go (hunk 2 above)
Comment (SoniaChoo) [reply db/rest.go]: updated.

File: http/restHandler.go (hunk 3 above)
Comment (SoniaChoo) [reply http/restHandler.go]: updated

**Review by ziyitony — APPROVED** [review]

**Pull request merged**